### PR TITLE
Added spellchecking in `docs` and `examples`

### PR DESCRIPTION
**Pull request by msaroufim**
Fixes problems with frequent typos in our examples and docs of https://github.com/pytorch/serve/issues/1331

h/t @d4l3k 

```
cd serve 
sh ts_scripts/spellcheck.sh 
```

@HamidShojanazeri @lxning could you please review the PR? It applies only to `examples/` and `docs/` 

After writing a new doc you would run the spellcheck script on your file and get a report of all typos

### CODE REVIEW

**Comment by d4l3k** [comment]
The spellcheck stuff hasn't really worked out terribly well for TorchX. It doesn't run on PRs and since it's a whitelist of words hard to maintain. It parses comments so you have to add every single function name to the whitelist. Probably a better way to handle that though

I occasionally run it but it's a lot of work each time

**Review by malfet — APPROVED** [review]

File: requirements/developer.txt [hunk 1]
```diff
@@ -10,4 +10,5 @@ grpcio
 protobuf
 grpcio-tools
 transformers==4.11.0
-pygit2==1.6.1
```
Comment (HamidShojanazeri) [inline requirements/developer.txt:13]: @msaroufim is installing "aspell" as mentioned in PR description still relevant?

Comment (HamidShojanazeri) [reply requirements/developer.txt:13]: @msaroufim seems it's still required to have "aspell" should we add it to the requirements?

 Regarding the vocab provided for spell checking, I wonder if lowercase of some of he words needs to be added as well? for example config, [here](https://gist.github.com/HamidShojanazeri/7433ef4f99a12b84be843aa3eb914714#file-gistfile1-txt-L27) in doc/serve.md

Comment (msaroufim) [reply requirements/developer.txt:13]: @HamidShojanazeri these should be fixed

**Review by HamidShojanazeri — APPROVED** [review]

**Pull request merged**